hack/lib/golang.sh

```diff
@@ -536,25 +536,42 @@ kube::golang::build_binaries_for_platform() {
       continue
     fi
 
-    # `go test -c` below directly builds the binary. It builds the packages,
-    # but it never installs them. `go test -i` only installs the dependencies
-    # of the test, but not the test package itself. So neither `go test -c`
-    # nor `go test -i` installs, for example, test/e2e.a. And without that,
-    # doing a staleness check on k8s.io/kubernetes/test/e2e package always
-    # returns true (always stale). And that's why we need to install the
-    # test package.
-    go install "${goflags[@]:+${goflags[@]}}" \
+    if [[ ! -n ${use_go_build:-} ]]; then
+      # `go test -c` below directly builds the binary. It builds the packages,
+      # but it never installs them. `go test -i` only installs the dependencies
+      # of the test, but not the test package itself. So neither `go test -c`
+      # nor `go test -i` installs, for example, test/e2e.a. And without that,
+      # doing a staleness check on k8s.io/kubernetes/test/e2e package always
+      # returns true (always stale). And that's why we need to install the
+      # test package.
+      go install "${goflags[@]:+${goflags[@]}}" \
+          -gcflags "${gogcflags}" \
+          -ldflags "${goldflags}" \
+          "${testpkg}"
+    fi
+
+    test_install=false
```

Comment:
This deserves at least as much comment as above.  I'm having a hard time figuring it out.